src/htr2hpc/train/run.py

```diff
@@ -147,6 +157,38 @@ def segmentation_training(self):
         os.chdir(self.orig_working_dir)
         self.upload_models()
 
+    def recognition_training(self):
+        # NOTE: this is nearly the same as segmentation_training method
+
+        # get absolute versions of these paths _before_ changing working directory
+        abs_training_data_dir = self.training_data_dir.absolute()
+        # input model is optional
+        abs_model_file = self.model_file.absolute() if self.model_file else None
+        abs_output_modelfile = self.output_modelfile.absolute()
+
+        # change directory to working directory, since by default,
+        # slurm executes the job from the directory where it was submitted
+        os.chdir(self.work_dir)
+
+        job_id = recognition_train(
+            abs_training_data_dir,
+            abs_output_modelfile,
+            abs_model_file,
+            self.num_workers,
+        )
+        self.monitor_slurm_job(job_id)
+        # change back to original working directory
+        os.chdir(self.orig_working_dir)
+        # look for and upload best model
+        best_model = upload_best_model(
+            self.api, self.output_modelfile.parent, self.training_mode
+        )
+        if best_model:
+            print(f"Uploaded {best_model} to eScriptorum")
+        else:
+            # possibly best model found but uploade failed?
```

Comment:
Connection issues (such as if the VM goes down for an inconvenient moment) could cause the upload to fail, so the fuller version of this should have a greater range of error handling. Not a priority, but not something we want to forget.

Reply:
yes, good point - could also happen when we're uploading lots of models as well. I'll create an issue for error handling and we can add more items to the list as we think of them.